### DISCUSSION TITLE
Improve docs with a example of how to do restore in case of disaster recovery

### DISCUSSION THREAD

**Discussion by midzelis** [zrepl/zrepl#730] [General]
I'm considering using this to replace my years long Crashplan cloud backups with a locally managed one. I plan on using encrypted backups from my local server(s) to a remote offsite machine (over VPN). 

I'm pretty familiar with ZFS, and have done manual snapshots/send/recvs many times. I'm less familiar with encryption. I plan on treating the remote as 'untrusted'. 

I noticed that the documentation does a great job explaining all the steps and configuration, but I don't see much explaining on how to do restores. 

I'm looking for 2 things: 
1) instructions on how to restore a snapshot that exists on remote (but has beed pruned from local) - to recover from `rm -rf` or other problems - the source hardware is otherwise completely intact and functional
2) instructions on how to recover in case of disaster: the original client machine is totally wiped out: hard drives in flames or under water. etc. Assume that a the OS was just reinstalled, and I need to recreate the zrepl install to match the previous - and then to restore everything from the remote back to local.  

I assume I would need to save a copy of at least
1) client encryption keys
2) client configuration

But, I'm probably missing things. I'll probably save these things in a secure note in 1password or on a usb stick in a fireproof safe. 

Thanks!

**Comment by jkotran** [comment]
@midzelis zrepl recovery may be performed outside of zrepl using zfs send / zfs receive from the backup to the primary. Another option is to rsync from backup to primary if you are comfortable losing snapshots or other zfs metadata.
